packages/exchange/src/exchange/providers/base.py

```diff
@@ -14,10 +15,19 @@ class Usage:
 
 
 class Provider(ABC):
+    PROVIDER_NAME: str
+    REQUIRED_ENV_VARS: list[str] = []
+
     @classmethod
     def from_env(cls: Type["Provider"]) -> "Provider":
         return cls()
 
+    @classmethod
+    def check_env_vars(cls: Type["Provider"], instructions_url: Optional[str] = None) -> None:
+        for env_var in cls.REQUIRED_ENV_VARS:
+            if env_var not in os.environ:
+                raise MissingProviderEnvVariableError(env_var, cls.PROVIDER_NAME, instructions_url)
```

Comment:
:nit: i feel we can improve the ux to show all of the errors in a single string. if we do 1 check at a time, a user would need to "discover" all of the missing vars looping over the list _n_ times

something like this maybe?
```python3
missing_vars = [x for x in cls.REQUIRED_ENV_VARS if x not in os.environ]

if missing_vars:
    print(f"Missing environment variables: {', '.join(missing_vars)}")
````